### PR TITLE
Deprecate unquoted-csv-writer and add in defusedcsv alternatives.

### PR DESCRIPTION
**Pull request by minusworld**
Addresses https://github.com/returntocorp/semgrep-rules/issues/2351

### CODE REVIEW

File: python/django/security/injection/csv-writer-injection.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - https://github.com/raphaelm/defusedcsv
+    - https://owasp.org/www-community/attacks/CSV_Injection
+    - https://web.archive.org/web/20220516052229/https://www.contextis.com/us/blog/comma-separated-vulnerabilities
+    technology:
+    - django
+    - python
+  mode: taint
+  pattern-sinks:
+  - patterns:
+    - pattern-inside: '$WRITER = csv.writer(...)
```
Comment (colleend) [inline python/django/security/injection/csv-writer-injection.yaml]: huh, nit but the formatting of this `pattern-inside` isn't like usual (`pattern-inside: |`)

File: python/django/security/injection/csv-writer-injection.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        $WRITER.$WRITE(...)
+
+        '
+    - pattern: $WRITER.$WRITE(...)
+    - metavariable-regex:
+        metavariable: $WRITE
+        regex: ^(writerow|writerows|writeheader)$
+  pattern-sources:
+  - patterns:
+    - pattern-inside: "def $FUNC(..., $REQUEST, ...):\n  ...\n"
```
Comment (colleend) [inline python/django/security/injection/csv-writer-injection.yaml]: same here

File: python/flask/security/injection/csv-writer-injection.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    - https://github.com/raphaelm/defusedcsv
+    - https://owasp.org/www-community/attacks/CSV_Injection
+    - https://web.archive.org/web/20220516052229/https://www.contextis.com/us/blog/comma-separated-vulnerabilities
+    technology:
+    - python
+    - flask
+  mode: taint
+  pattern-sinks:
+  - patterns:
+    - pattern-inside: '$WRITER = csv.writer(...)
```
Comment (colleend) [inline python/flask/security/injection/csv-writer-injection.yaml]: same here

File: python/flask/security/injection/csv-writer-injection.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          - pattern: flask.request.headers[...]
+          - pattern: flask.request.data
+          - pattern: flask.request.full_path
+          - pattern: flask.request.url
+          - pattern: flask.request.json
+          - pattern: flask.request.get_json()
+          - pattern: flask.request.view_args.get(...)
+          - pattern: flask.request.view_args[...]
+      - patterns:
+        - pattern: "@$APP.route($ROUTE, ...)\ndef $FUNC(..., $ROUTEVAR, ...):\n  ...\n"
```
Comment (colleend) [inline python/flask/security/injection/csv-writer-injection.yaml]: could also be `pattern: |`

**Review by colleend — DISMISSED** [review]
seems like there's some formatting differences, but otherwise rules look good

File: python/flask/security/injection/csv-writer-injection.yaml (hunk 4 above)
Comment (kurt-r2c) [reply python/flask/security/injection/csv-writer-injection.yaml]: the script that is generating these almost certainly needs to override the string presenter:

```python
# https://stackoverflow.com/questions/8640959/how-can-i-control-what-scalar-form-pyyaml-uses-for-my-data/15423007#15423007
def str_presenter(dumper, data):
  if '\n' in data:  # check for multiline string
    return dumper.represent_scalar('tag:yaml.org,2002:str', data, style='|')
  return dumper.represent_scalar('tag:yaml.org,2002:str', data)
...
yaml.add_representer(str, str_presenter)
```

File: python/django/security/injection/csv-writer-injection.yaml (hunk 1 above)
Comment (p4p3r) [reply python/django/security/injection/csv-writer-injection.yaml]: Addressed in 2f50dbd1a20753034d4a10ebd0e4a855659ccfc5

File: python/django/security/injection/csv-writer-injection.yaml (hunk 2 above)
Comment (p4p3r) [reply python/django/security/injection/csv-writer-injection.yaml]: Addressed in 2f50dbd1a20753034d4a10ebd0e4a855659ccfc5

File: python/flask/security/injection/csv-writer-injection.yaml (hunk 3 above)
Comment (p4p3r) [reply python/flask/security/injection/csv-writer-injection.yaml]: Addressed in 2f50dbd1a20753034d4a10ebd0e4a855659ccfc5

File: python/flask/security/injection/csv-writer-injection.yaml (hunk 4 above)
Comment (p4p3r) [reply python/flask/security/injection/csv-writer-injection.yaml]: Addressed in 2f50dbd1a20753034d4a10ebd0e4a855659ccfc5

**Review by p4p3r — APPROVED** [review]

**Pull request merged**